app/src/main/java/org/oppia/app/player/state/InteractionAdapter.kt

```diff
@@ -0,0 +1,142 @@
+import androidx.databinding.DataBindingUtil
+import androidx.databinding.ViewDataBinding
+import androidx.databinding.library.baseAdapters.BR
+import androidx.recyclerview.widget.RecyclerView
+import kotlinx.android.synthetic.main.item_selection_interaction_items.view.*
+import kotlinx.android.synthetic.main.multiple_choice_interaction_items.view.*
+import org.oppia.app.R
+import org.oppia.app.databinding.ItemSelectionInteractionItemsBinding
+import org.oppia.app.databinding.MultipleChoiceInteractionItemsBinding
+import org.oppia.app.model.InteractionObject
+import org.oppia.app.player.state.listener.InteractionAnswerRetriever
+import org.oppia.app.player.state.listener.InteractionListener
+import org.oppia.util.parser.HtmlParser
+
+const val VIEW_TYPE_MULTIPLE_CHOICE = 1
+const val VIEW_TYPE_ITEM_SELECTION = 2
+
+/** Adapter to bind the interactions to the [RecyclerView]. It handles MultipleChoiceInput and ItemSelectionInput interaction views. */
+class InteractionAdapter(
+  private val htmlParserFactory: HtmlParser.Factory,
+  private val entityType: String,
+  private val explorationId: String,
+  val itemList: Array<String>?,
+  private val interactionId: String
+  ) : RecyclerView.Adapter<RecyclerView.ViewHolder>(), InteractionAnswerRetriever {
+
+  private var mSelectedItem = -1
+
+  private var selectedAnswerIndex = -1
+
+  override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
+    return when (viewType) {
+      VIEW_TYPE_MULTIPLE_CHOICE -> {
+        val inflater = LayoutInflater.from(parent.getContext())
+        val binding =
+          DataBindingUtil.inflate<MultipleChoiceInteractionItemsBinding>(
+            inflater,
+            R.layout.multiple_choice_interaction_items,
+            parent,
+            false
+          )
+        MultipleChoiceViewHolder(binding)
+      }
+      VIEW_TYPE_ITEM_SELECTION -> {
+        val inflater = LayoutInflater.from(parent.getContext())
+        val binding =
+          DataBindingUtil.inflate<ItemSelectionInteractionItemsBinding>(
+            inflater,
+            R.layout.item_selection_interaction_items,
+            parent,
+            false
+          )
+        ItemSelectionViewHolder(binding)
+      }
+      else -> throw IllegalArgumentException("Invalid view type")
+    }
+  }
+
+  override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
+    when (holder.itemViewType) {
+      VIEW_TYPE_MULTIPLE_CHOICE -> (holder as MultipleChoiceViewHolder).bind(
+        itemList!!.get(position),
+        position,
+        mSelectedItem
+      )
+      VIEW_TYPE_ITEM_SELECTION -> (holder as ItemSelectionViewHolder).bind(
+        itemList!!.get(position),
+        position,
+        mSelectedItem
+      )
+    }
+  }
+
+  // Determines the appropriate ViewType according to the interaction type.
+  override fun getItemViewType(position: Int): Int {
+    return if (interactionId.equals("ItemSelectionInput")) {
```

Comment:
change condition to `interactionId == "ItemSelectionInput"`

Reply:
done.